src/Rules/Test.hs

```diff
@@ -63,13 +64,24 @@ testRules = do
         -- Execute the test target.
         buildWithCmdOptions env $ target (vanillaContext Stage2 compiler) RunTest [] []
 
+-- | Build extra programs required by testsuite
+needTestsuiteBuilders :: Action ()
+needTestsuiteBuilders = do
+    targets <- mapM (f Stage1) =<< testsuitePackages
+    need targets
+  where
+    f :: Stage -> Package -> Action FilePath
+    f stage pkg | isLibrary pkg = pkgConfFile (Context stage pkg (read "v"))
+                | otherwise     = programPath =<< programContext stage pkg
+
 needTestBuilders :: Action ()
 needTestBuilders = do
     needBuilder $ Ghc CompileHs Stage2
     needBuilder $ GhcPkg Update Stage1
     needBuilder Hp2Ps
```

Comment:
`Hp2Ps` is already included into `testsuitePackages`, so I guess we can drop this line?